src/components/swapv2/LimitOrder/useBaseTradeInfo.ts

```diff
@@ -2,57 +2,88 @@ import { Currency, CurrencyAmount, Price, TokenAmount } from '@kyberswap/ks-sdk-
 import axios from 'axios'
 import JSBI from 'jsbi'
 import { stringify } from 'querystring'
+import { useRef } from 'react'
 import useSWR from 'swr'
 
 import { ETHER_ADDRESS, ZERO_ADDRESS, sentryRequestId } from 'constants/index'
 import { NETWORKS_INFO } from 'constants/networks'
 import { useActiveWeb3React } from 'hooks'
 import { tryParseAmount } from 'state/swap/hooks'
 
+type BaseTradeInfo = {
+  price: Price<Currency, Currency>
+  amountInUsd: number
+  amountOutUsd: number
+  routerAddress: string
+}
+
+const MAX_RETRY_COUNT = 2
+
 // 1 knc = ?? usdt
 export default function useBaseTradeInfo(currencyIn: Currency | undefined, currencyOut: Currency | undefined) {
   const { account, chainId } = useActiveWeb3React()
   const tokenInAddress = currencyIn?.isNative ? ETHER_ADDRESS : currencyIn?.wrapped.address ?? ''
   const tokenOutAddress = currencyOut?.isNative ? ETHER_ADDRESS : currencyOut?.wrapped.address ?? ''
   const amountIn = tryParseAmount('1', currencyIn)
 
-  const { data, isValidating } = useSWR(
-    tokenInAddress && tokenOutAddress && chainId
+  const getApiUrl = () => {
+    return tokenInAddress && tokenOutAddress && chainId
       ? `${NETWORKS_INFO[chainId].routerUri}?${stringify({
           tokenIn: tokenInAddress.toLowerCase(),
           tokenOut: tokenOutAddress.toLowerCase(),
           amountIn: amountIn?.quotient?.toString() ?? '',
           to: account ?? ZERO_ADDRESS,
         })}`
-      : null,
-    async (url: string) => {
-      if (!currencyOut || !currencyIn || !url) return
+      : null
+  }
+  const fetchData = async (url: string | null): Promise<BaseTradeInfo | undefined> => {
+    if (!currencyOut || !currencyIn || !url) return
+    const { data } = await axios.get(url, {
+      headers: {
+        'X-Request-Id': sentryRequestId,
+        'Accept-Version': 'Latest',
```

Comment:
I think this isnt supported?

Reply:
I see kyberswap currently uses it in API `route/encode` (aggregator.ts bestTradeExactIn function)

Reply:
It is safe to remove.

Reply:
We do not use `Accept-Version` anywhere.

Reply:
this is still api route/encode, I think the backend need it for tracking, ...
if you confirm it is not necessary. I will remove.